pkg/k8sutil/exec.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8sutil
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+var (
+	OperatorNamespace    string // operator namespae
+	CephClusterNamespace string // Cephcluster namespace
+)
+
+func RunCommandInOperatorPod(cmd string, args []string, operatorNamespace, clusterNamespace string) {
+	// 1. Create Kubernetes Client
+	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{},
+	)
+	rest, err := kubeconfig.ClientConfig()
```

Comment:
Seems like these lines 42-49 are duplicated from what is in GetKubeClient(). Instead of duplicating, how about a different helper method instead of GetKubeClient() that gives back the `rest` and `client`? Or some other refactoring.

Reply:
In the future we may also want to unit test commands where we can mock K8s and the exec commands, like we do in the Rook core. So I'd suggest that this method should take the kubeconfig as a parameter, then the core implementation takes parameters for the context that can be mocked. The kubeconfig would likely be initialized at the top layer under the cmd package.

Reply:
now, I'm returning the rest and client from `GetKubeClient` only.
I'm not sure about second part could you refer some example